kong/admin_service.go

```diff
@@ -0,0 +1,174 @@
+package kong
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+// AdminService handles Admins in Kong.
+type AdminService service
+
+// Invite creates an Admin in Kong.
+func (s *AdminService) Invite(ctx context.Context,
+	admin *Admin) (*Admin, error) {
+
+	if admin == nil {
+		return nil, errors.New("cannot create a nil admin")
+	}
+
+	endpoint := "/admins"
+	method := "POST"
+	req, err := s.client.NewRequest(method, endpoint, nil, admin)
+	if err != nil {
+		return nil, err
+	}
+
+	var createdAdmin struct {
+		Admin Admin `json:"admin,omitempty" yaml:"admin,omitempty"`
+	}
```

Comment:
This bit seems like a hack in the API but the smell is Kong Enterprise and not this patch. Leaving a note for future reference, no changes required.